bigquery_etl/view/generate_views.py

```diff
@@ -77,81 +80,89 @@ def main():
             views[view] = []
         views[view].append(full_table_id)
 
-    create_views_if_not_exist(client, views, args.exclude, args.sql_dir)
-
-
-def create_views_if_not_exist(client, views, exclude, sql_dir):
-    """Create views unless a local file for creating the view exists."""
-    for view, tables in views.items():
-        if any(fnmatchcase(pattern, view) for pattern in exclude):
-            logging.info("skipping table: matched by exclude pattern: {view}")
-            continue
-        if view.endswith("_"):
-            # A trailing '_' confuses the logic here of parsing versions,
-            # and likely indicates that the table is somehow private, so
-            # we ignore it.
-            logging.info("skipping table ending in _: {view}")
-            continue
-
-        version = max(
-            int(match.group()[2:])
-            for table in tables
-            for match in [VERSION_RE.search(table)]
-            if match is not None
+    with ThreadPool(args.parallelism) as pool:
+        pool.starmap(
+            partial(
+                create_view_if_not_exists,
+                client,
+                exclude=args.exclude,
+                sql_dir=args.sql_dir,
+            ),
+            views.items()
         )
 
-        project, dataset, viewname = view.split(".")
-        target = f"{view}_v{version}"
-        view_dataset = dataset.rsplit("_", 1)[0]
-        full_view_id = ".".join([project, view_dataset, viewname])
-        target_file = os.path.join(sql_dir, project, view_dataset, viewname, "view.sql")
-
-        if not os.path.exists(target_file):
-            # We put this BQ API all inside the conditional to speed up execution
-            # in the case target files already exist.
-            table = client.get_table(target)
-            replacements = ["mozfun.norm.metadata(metadata)" " AS metadata"]
-            # Add special replacements for Glean pings.
-            schema_id = table.labels.get("schema_id", None)
-            if schema_id == "glean_ping_1":
+
+def create_view_if_not_exists(client, view, tables, exclude, sql_dir):
```

Comment:
We've changed the signature here, but the body of the function is unchanged other than missing one level of indent since the function now operates on a single view rather than iterating over an object. The iteration now happens in the `pool.starmap` call.